packages/react-components/react-provider/src/components/FluentProvider/useFluentProviderThemeStyleTag.ts

```diff
@@ -62,7 +62,7 @@ export const useFluentProviderThemeStyleTag = (options: Pick<FluentProviderState
       : '';
   }, [theme]);
 
-  const rule = `.${styleTagId} { ${cssVarsAsString} }`;
+  const rule = `.${styleTagId.replace(/:/g, '\\:')} { ${cssVarsAsString} }`;
```

Comment:
> Note: React's useId hook uses : for its generated id, this is a problem for us since we use ids as class names when loading up our theme css vars. To fix this issue, we have to replace : with another string, in this case --.

This does not match PR's description :) Can you please update the description and a small comment there? 🙏